### PR TITLE
Minor sdba fixes

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [ ] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #xyz
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [ ] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 

Small tweaks and workarounds that are needed for proper usage of the sdba module.

1) All-NaN slices : `interp_on_quantiles` would fail on all-nan slices, I added a check.

2) Chunking after grouping ops. Dask arrays are returned chunked along the grouped dimension after `groupby` calls, but  sdba was written assuming the main dimension (time) has only one chunk. I added checks and rechunking in `Group.apply`. Moreover,  `add_cyclic_bounds` and `add_endpoints` need to rechunk their output to 1 chunk if the input has only one chunk. If not, sdba also fails.
Here I chose to fix the problems by rechunking the arrays just after the operation that added problematic chunks. I could have done the opposite: rechunk the arrays just before the operation that needs individual chunks (interps and apply_ufuncs)... Any opinions on this?

3) The  `adjust` methods were adding a boolean `bias_adjusted` attribute, but netCDF4 was failing when writing this to disk. I modified by updating the history of `scen`, appending the one of `sim`.

* **Does this PR introduce a breaking change?** 
No, since nobody uses `sdba` except me.

* **Other information**:

### CODE REVIEW

File: xclim/sdba/adjustment.py [hunk 1]
```diff
@@ -88,7 +89,9 @@ def adjust(self, sim: DataArray, **kwargs):
                 stacklevel=4,
             )
         scen = self._adjust(sim, **kwargs)
-        scen.attrs["bias_adjusted"] = True
+        scen.attrs["history"] = update_history(
+            f"Bias-adjusted with method {str(self)}", sim
+        )
```
Comment (huard) [inline xclim/sdba/adjustment.py:94]: :+1:

File: xclim/sdba/base.py [hunk 2]
```diff
@@ -263,6 +274,10 @@ def apply(
                 out = out.squeeze(self.dim, drop=True)  # .drop_vars(self.dim)
             else:
                 out = out.sortby(self.dim)
+                if out.chunks is not None and not dim_is_chunked:
+                    out = out.chunk({self.dim: -1})
+        elif self.prop in out.dims and out.chunks is not None:
+            out = out.chunk({self.prop: -1})
```
Comment (huard) [inline xclim/sdba/base.py:280]: Add some comments on why this is needed.

File: xclim/sdba/utils.py [hunk 3]
```diff
@@ -276,6 +276,8 @@ def add_cyclic_bounds(da: xr.DataArray, att: str, cyclic_coords: bool = True):
         vals[-1] = vals[-2] + diff[-1]
         qmf = qmf.assign_coords({att: vals})
         qmf[att].attrs.update(da.coords[att].attrs)
+    if da.chunks is not None and len(da.chunks[da.get_axis_num(att)]) == 1:
```
Comment (huard) [inline xclim/sdba/utils.py:279]: Add explanation comment.

File: xclim/sdba/utils.py [hunk 4]
```diff
@@ -416,8 +421,13 @@ def _interp_quantiles_1D(newx, oldx, oldy):
     # else:
 
     def _interp_quantiles_2D(newx, newg, oldx, oldy, oldg):
+        print('|')
```
Comment (huard) [inline xclim/sdba/utils.py:424]: remove

Comment (aulemahal) [reply xclim/sdba/utils.py:424]: Argh, now you see my secret debugging methods!

**Pull request merged**